### ISSUE TITLE
Error identifying terminal

### ISSUE THREAD

**Issue by jooyoungseo** [REditorSupport/vscode-R#1459]
Type: <b>Bug</b>

It looks like there is an error identifying my R terminal when executing R command. I am attaching the screen shot below.

![image](https://github.com/REditorSupport/vscode-R/assets/19754711/6d4663fa-9df8-47f5-a412-2d14a89f14ed)

Extension version: 2.8.2
VS Code version: Code - Insiders 1.85.0-insider (af28b32d7e553898b2a91af498b1fb666fdebe0c, 2023-12-06T11:15:22.766Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz (8 x 2611)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.71GB (6.17GB free)|
|Process Argv|C:\\Users\\jseo1005\\OneDrive - University of Illinois - Urbana\\Desktop\\source.R --crash-reporter-id b05b88e5-8894-4031-ae34-fa034ebddea9|
|Screen Reader|yes|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
py29gd2263:30784851
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
a9j8j154:30646983
showlangstatbar:30737417
fixshowwlkth:30771523
showindicator:30805243
pythongtdpath:30726887
i26e3531:30792625
welcomedialog:30812478
pythonnosmt12:30779711
pythonidxpt:30768918
pythonnoceb:30776497
asynctok:30898717
dsvsc013:30777762
dsvsc014:30777825
pythonmpswarning:30901777
dsvsc015:30821418
pythontestfixt:30866404
pythonregdiag2:30902439
pyreplss1:30879911
pythonmypyd1:30859725
pythoncet0:30859736
pythontbext0:30879054
accentitlementst:30870582
dsvsc016:30879898
dsvsc017:30880771
dsvsc018:30880772
aa_t_chat:30882232

```

</details>

**Comment by camilogarciabotero** [comment]
Hi,

I have a similar issue when trying to attach R to the terminal. I am now with the released version of VS code (v1.85.0). The output says:

```bash
.vsc.attach()
Fatal error: cannot create 'R_TempDir'
```

I am posting here because it looks related. Yet I am working from a remote cluster with `CentOS v8`. If more info is needed I will be looking forward to providing it.

**Comment by tomasnobrega** [comment]
Hi, were you able to fix it? I am running into exactly the same error message

**Comment by andycraig** [comment]
Previously this error had its root cause in a bug in VS Code itself. https://github.com/REditorSupport/vscode-R/issues/385 I wonder if that bug has reappeared.

@jooyoungseo Thank you for reporting this!

@camilogarciabotero I think that's probably a different bug.

@tomasnobrega Would you be able to post the screenshot too? Thank you.

**Comment by xinzhuohkust** [comment]
![image](https://github.com/REditorSupport/vscode-R/assets/99251348/9d5c547f-0b38-4616-9597-c84106ec9051)

Got the same error. 

<details>
Version: 1.87.0 (user setup)
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:41:44.469Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.19045
</details>

**Comment by andycraig** [comment]
@xinzhuohkust Thank you!

@jooyoungseo @tomasnobrega @xinzhuohkust  Have you noticed any pattern to when this happens? After using VS Code for a while, after closing several terminals, only when using remote etc.?

**Comment by camilogarciabotero** [comment]
My issue was related to intermediate files in the `temp`. I found this [StackOverflow answer](https://stackoverflow.com/questions/6307836/fatal-error-cannot-mkdir-r-tempdir/71841130#71841130). And work for me:

```bash
find . -type d -name 'Rtmp*' -exec rm -r -v {} \;
```

**Comment by tomasnobrega** [comment]
I get the same error as @xinzhuohkust:
<img width="551" alt="image" src="https://github.com/REditorSupport/vscode-R/assets/76699840/d5048fc0-4e6e-498e-80d1-55259a097172">

After I kill the currently active R terminal and then try to run a line from my R script using "Command + Enter," the command does not execute as expected. However, if I manually open a new terminal, type "R" to start an R session, and then use "Command + Enter," I can run lines from my script without any problems. I'm not sure why this is happening, but it appears that the automatic creation of a new R terminal after killing the previous one is not functioning correctly.

**Comment by tomasnobrega** [comment]
Not sure if anything was made, but this issue is not happening with me anymore. I didn't change anything on my end, not that I am aware anyways.

**Comment by tomasnobrega** [comment]
Hi guys, I am facing this issue again. I know this is probably related to some user configuration as this is not showing for other people. I have spent way too much time trying to figure this out. I wonder if I can access the contents of `vscode.window.terminals` somehow to try to figure out which terminals are conflicting in my machine. Aparently I have 2 terminals and this is causing the issue.

**Comment by andycraig** [comment]
This should be solved now thanks to #1511. Thank you @tomasnobrega!
